docs/api.asciidoc

```diff
@@ -0,0 +1,159 @@
+
+[break]
+[[client-api-capture-exception]]
+==== `Client.capture_message()`
+
+Captures a message with optional added contextual data.
+
+[source,python]
+----
+client.capture_message('Billing process succeeded.')
+----
+
+[break]
+===== Arguments
+
+ * `message`: The message as a string.
+ * `param_message`: Alternatively, a parametrized message as a dictionary.
+    The dictionary contains two values: `message`, and `params`.
+    This allows the APM server to group messages together that have share the same
+    parametrized message. Example:
++
+[source,python]
+----
+client.capture_message(param_message={
+    'message': 'Billing process for %s succeeded. Amount: %s',
+    'params': (customer.id, order.total_amount),
+})
+----
++
+ * `stack`: If set to `True` (the default), a stacktrace from the call site will be captured.
+ * `exc_info`: A `(type, value, traceback)` tuple as returned by https://docs.python.org/3/library/sys.html#sys.exc_info[`sys.exc_info()`]. If not provided, it will be captured automatically, if `capture_message()` was called in an `except` block.
+ * `date`: A `datetime.datetime` object representing the occurrence time of the error. If left empty, it defaults to `datetime.datetime.utcnow()`.
+ * `context`: A dictionary with contextual information. This dictionary must follow the 
+    https://www.elastic.co/guide/en/apm/server/current/error-api.html#error-context-schema[Context] schema definition.
+ * `custom`: A dictionary of custom data you want to attach to the event.
+
+Returns the id of the message as a string.
+
+[[transaction-api]]
+=== Transaction API
+
+[break]
+[[client-api-begin-transaction]]
+==== `Client.begin_transaction()`
+Begin tracking a transaction.
+Should be called e.g. at the beginning of a request or when starting a background task.
+
+[break]
+===== Arguments
+
+ * `transaction_type`: A string describing the type of the transaction, e.g. `'request'` or `'celery'`.
+
+[break]
+[[client-api-end-transaction]]
+==== `Client.end_transaction()`
+End tracking the transaction.
+Should be called e.g. at the end of a request or when ending a background task.
+
+[break]
+===== Arguments
+
+ * `name`: A string describing the name of the transaction, e.g. `process_order`.
+   This is typically the name of the view/controller that handles the request, or the route name.
+ * `result`: A string describing the result of the transaction.
+   This is typicaly the HTTP status code, or e.g. `'success'` for a background task.
+
+[break]
+[[api-elasticapm-instrument]]
+==== `elasticapm.instrument()`
+
+Instruments libraries automatically.
+This includes a wide range of standard library and 3rd party modules.
+A list of instrumented modules can be found in `elasticapm.intrumentation.register`.
+This function should be called as early as possibly in the startup of your application.
+For supported frameworks, this is called automatically.
+
+[break]
+[[api-set-transaction-name]]
+==== `elasticapm.set_transaction_name()`
```

Comment:
Why is `begin_transaction` and `end_transaction` a method of `client` byt `set_transaction_name` and `set_transaction_data` are not?

Reply:
in most places you'd use `set_transaction_name` & Co, you don't have a `client` instance or the current transaction. With this, we can put the difficulty of getting the current transaction into our code, instead of leaving it to the user to acquire the current transaction somehow.

At the moment, the current transaction is stored on a thread local. But that only works for singlethreaded code, so we probably need to change that in the future. The way it is now, that change would be nicely encapsulated.

Reply:
How can you not have a current transaction when you want to override the name of the current transaction? 
I ofc don't have the insights into the agent, but from a pure user perspective I'd somwhow expect that `set_transaction_name` and `begin/stop_transaction` is within the same context. 
But this might be somehow out of scope for this PR anyways, just realized that when looking at the docs.